### PR TITLE
Extended `MetricsLambda` to update its underlying metrics

### LINKED ISSUE #1921: Make `MetricsLambda` update it's underlying metrics
## 🚀 Feature

As title.

The current state is that we cannot use `MetricsLambda` as a standard metric because it doesn't update/reset it's underlying metrics.
Adding this will expand the scope and usability of `MetricsLambda`

I've already an idea how to implement this based on our discussion https://github.com/pytorch/ignite/pull/1887#discussion_r610902667

So @vfdev-5 please assign the issue to me.

### PR DESCRIPTION
**Pull request by KickItLikeShika**
Fixes #1921 

Implemented the update method to make metrics lambda able to update its underlying methods, and I separated the attach from reset/update/compute API, then implemented the necessary tests.

Check list:

- [x] New tests are added (if a new feature is added)
- [x] New doc strings: description and/or example code are in RST format
- [x] Documentation is updated (if required)

### CODE REVIEW

File: ignite/metrics/metrics_lambda.py [hunk 1]
```diff
@@ -105,6 +114,10 @@ def _internal_attach(self, engine: Engine, usage: MetricUsage) -> None:
                     engine.add_event_handler(usage.ITERATION_COMPLETED, metric.iteration_completed)
 
     def attach(self, engine: Engine, name: str, usage: Union[str, MetricUsage] = EpochWise()) -> None:
+        if self._updated:
+            raise ValueError(
+                "The underlying metrics are already updated, " "can't attach while using reset/update/compute API."
```
Comment (vfdev-5) [inline ignite/metrics/metrics_lambda.py:119]:
```suggestion
                "The underlying metrics are already updated, can't attach while using reset/update/compute API."
```

**Review by vfdev-5 — COMMENTED** [review]
Thanks a lot for the PR @KickItLikeShika

File: ignite/metrics/metrics_lambda.py (hunk 1 above)
Comment (KickItLikeShika) [reply ignite/metrics/metrics_lambda.py:119]: `black` magic again :)))

**Review by vfdev-5 — APPROVED** [review]
LGTM, thanks a lot @KickItLikeShika !

**Pull request merged**